packages/contact-center/task/tests/helper.ts

```diff
@@ -0,0 +1,215 @@
+
+    waitFor(() => {
+      expect(result.current.currentTask).toBe(taskMock);
+      expect(ccMock.on).toHaveBeenCalledWith('TASK_INCOMING', expect.any(Function));
+    });
+  });
+
+  it('should register task events for the current task', async () => {
+    const {result} = renderHook(() =>
+      useIncomingTask({cc: ccMock, onAccepted, onDeclined, selectedLoginOption: 'BROWSER'})
+    );
+
+    // Simulate an incoming task
+    act(() => {
+      ccMock.on.mock.calls[0][1](taskMock); // Trigger TASK_INCOMING event
+    });
+
+    waitFor(() => {
+      expect(taskMock.on).toHaveBeenCalledWith('TASK_ASSIGNED', expect.any(Function));
+      expect(taskMock.on).toHaveBeenCalledWith('TASK_END', expect.any(Function));
+      expect(taskMock.on).toHaveBeenCalledWith('TASK_UNASSIGNED', expect.any(Function));
+      expect(taskMock.on).toHaveBeenCalledWith('TASK_MEDIA', expect.any(Function));
+    });
+  });
+
+  it('should clean up task events on task change or unmount', async () => {
+    const {result, unmount} = renderHook(() =>
+      useIncomingTask({cc: ccMock, onAccepted, onDeclined, selectedLoginOption: 'BROWSER'})
+    );
+
+    // Simulate an incoming task
+    act(() => {
+      ccMock.on.mock.calls[0][1](taskMock); // Trigger TASK_INCOMING event
+    });
+
+    // Simulate unmount
+    unmount();
+
+    waitFor(() => {
+      expect(taskMock.off).toHaveBeenCalledWith('TASK_ASSIGNED', expect.any(Function));
+      expect(taskMock.off).toHaveBeenCalledWith('TASK_END', expect.any(Function));
+      expect(taskMock.off).toHaveBeenCalledWith('TASK_UNASSIGNED', expect.any(Function));
+      expect(taskMock.off).toHaveBeenCalledWith('TASK_MEDIA', expect.any(Function));
+      expect(ccMock.off).toHaveBeenCalledWith('TASK_INCOMING', expect.any(Function));
+    });
+  });
+
+  it('should call onAccepted callback when task is accepted', async () => {
+    const {result} = renderHook(() =>
+      useIncomingTask({cc: ccMock, onAccepted, onDeclined, selectedLoginOption: 'BROWSER'})
+    );
+
+    // Simulate an incoming task
+    act(() => {
+      ccMock.on.mock.calls[0][1](taskMock); // Trigger TASK_INCOMING event
+    });
+
+    // Accept the task
+    act(() => {
+      result.current.accept();
+    });
+
+    waitFor(() => {
+      expect(onAccepted).toHaveBeenCalled();
+      expect(taskMock.accept).toHaveBeenCalledWith('interaction1');
+    });
+  });
+
+  it('should call onDeclined callback when task is declined', async () => {
+    const {result} = renderHook(() =>
+      useIncomingTask({cc: ccMock, onAccepted, onDeclined, selectedLoginOption: 'BROWSER'})
+    );
+
+    // Simulate an incoming task
+    act(() => {
+      ccMock.on.mock.calls[0][1](taskMock); // Trigger TASK_INCOMING event
+    });
+
+    // Decline the task
+    act(() => {
+      result.current.decline();
+    });
+
+    waitFor(() => {
+      expect(onDeclined).toHaveBeenCalled();
+      expect(taskMock.decline).toHaveBeenCalledWith('interaction1');
+    });
+  });
```

Comment:
We can have another test, when there is no onAccept and onDecline, we don't call it. This was an issue in station-login which caused errors.